### PR TITLE
Main menu screen +  wave logic

### LINKED ISSUE #15: Canon doesn't shoot if switching rapidly between two targets
We could imagine that the canon will fire if the closest target is less than X° away than the canon, with X being 5° for example?

### LINKED ISSUE #19: Wave logic
- [ ] On start, leave some time before enemy spawns (throwaway timer)
- [ ] Use resources to dictate how waves behave
  - How many enemies
  - Delay between spawns
  - Life and attack of enemy?
- [ ] When wave is done, show in UI, let some time before next waves starts
- [ ] When every wave is done, show game over screen (but as a win)

### PR DESCRIPTION
**Pull request by Kinsteen**
Closes #19, closes #15

Maybe an ordered list of screen on top of the gui would be good? Easier management of multiple windows/screens on top of each other

### CODE REVIEW

File: scenes/PlayMenu.gd [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+signal quit_button_pressed
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	pass
```
Comment (PaulLuperini) [inline scenes/PlayMenu.gd:13]: You could remove `_ready` and `_process` as they are not used.

File: scenes/GameScene.gd [hunk 2]
```diff
@@ -1,13 +1,15 @@
 extends Node2D
 
+@onready var enemy_spawner = $EnemySpawner as EnemySpawner
+
+const WAVE_1 = preload("res://resources/Wave1.tres")
```
Comment (PaulLuperini) [inline scenes/GameScene.gd:5]: Are these variables used? I removed them and saw no issue or difference in gameplay.

**Review by PaulLuperini — CHANGES_REQUESTED** [review]
Nice work :)

However, having randomness in wave spawns seems weird since it means that waves are not reproducible 🤔

**Comment by Kinsteen** [comment]
Does it matter if waves are not reproducible? I did it just to test stuff, but we could remove it

**Comment by PaulLuperini** [comment]
No, indeed, it doesn't matter right now. We can remove it at a later stage.

**Pull request merged**